### ISSUE TITLE
Multi-Join

### ISSUE THREAD

**Issue by ViceIce** [andrei-markeev/camljs#8]
Hi,

I would like to join three lists like in [Joins and Projections in Queries](https://msdn.microsoft.com/en-us/library/office/ee539975(v=office.14).aspx?f=255&mspperror=-2147217396#Anchor_1):

`Orders > Customers > Cities`

It looks like this is currently not possible.

Regards
Michael

**Comment by andrei-markeev** [comment]
Hi Michael,

Thanks for reporting!
I implemented this enhancement in [camljs 2.8.2](https://github.com/andrei-markeev/camljs/releases/tag/2.8.2) ([camljs-console 1.0.9](https://chrome.google.com/webstore/detail/camljs-console/ohjcpmdjfihchfhkmimcbklhjdphoeac)). NuGet package also updated, but it might take a bit of time to update other sources (e.g. DefinitelyTyped).

So for example the following **CamlBuilder** query:
```js
var query = new CamlBuilder()
    .View(["Title", "CustomerCity"])
    .LeftJoin("CustomerName", "customers")
    .LeftJoin("CityName", "customerCities", "customers")
    .Select("Title", "CustomerCity")
    .Query()
    .ToString();
```

will result in:
```xml
<View>
    <ViewFields>
        <FieldRef Name="Title" />
        <FieldRef Name="CustomerCity" />
    </ViewFields>
    <Joins>
        <Join Type="LEFT" ListAlias="customers">
            <Eq>
                <FieldRef Name="CustomerName" RefType="ID" />
                <FieldRef Name="ID" List="customers" />
            </Eq>
        </Join>
        <Join Type="LEFT" ListAlias="customerCities">
            <Eq>
                <FieldRef Name="CityName" RefType="ID" List="customers" />
                <FieldRef Name="ID" List="customerCities" />
            </Eq>
        </Join>
    </Joins>
    <ProjectedFields>
        <Field ShowField="Title" Type="Lookup" Name="CustomerCity" List="customerCities" />
    </ProjectedFields>
    <Query />
</View>
```

(example is very similar to the one in the article, https://msdn.microsoft.com/en-us/library/office/ee539975(v=office.14).aspx?f=255&mspperror=-2147217396#Anchor_1)

**Comment by ViceIce** [comment]
One problem:

```csharp
var query = new CamlBuilder()
    .View(["Title", "CustomerName", "CustomerCity"])
    .LeftJoin("CustomerName", "customers")
    .Select("Name", "CustomerName")
    .LeftJoin("CityName", "customerCities", "customers")
    .Select("Title", "CustomerCity")
    .Query()
    .ToString();
```

Produces:
```xml
<View>
    <ViewFields>
        <FieldRef Name="Title" />
        <FieldRef Name="CustomerName" />
        <FieldRef Name="CustomerCity" />
    </ViewFields>
    <Joins>
        <Join Type="LEFT" ListAlias="customers">
            <Eq>
                <FieldRef Name="CustomerName" RefType="ID" />
                <FieldRef Name="ID" List="customers" />
            </Eq>
        </Join>
        <Join Type="LEFT" ListAlias="customerCities">
            <Eq>
                <FieldRef Name="CityName" RefType="ID" />
                <FieldRef Name="ID" List="customerCities" />
            </Eq>
        </Join>
    </Joins>
    <ProjectedFields>
        <Field ShowField="Name" Type="Lookup" Name="CustomerName" List="customers" />
        <Field ShowField="Title" Type="Lookup" Name="CustomerCity" List="customerCities" />
    </ProjectedFields>
    <Query />
</View>
```

The list attribute for CityName at customerCities is missing.

The `JoinManager.ProjectField` return the original view, which does not accept the `fromList` parameter,
but the return type `IProjectableView` allows the `fromList` param.
